vmaas/reposcan/database/csaf_store.py

```diff
@@ -1,178 +1,378 @@
+        for cve, products in csaf_cves.items():
+            products_copy = deepcopy(products)  # only for logging of failed cves
+            try:
+                self._update_product_ids(products)
+                self._insert_missing_products(products)
+                self._remove_cves(cve, products)
+                self._insert_cves(cve, products)
+            except CsafStoreSkippedCVE as exc:
+                self.logger.exception("Skipping cve: %s reason: %s", cve, exc)
+                self.logger.debug("Skipping cve: %s products: %s reason: %s", cve, products_copy, exc)
+            except CsafStoreException:
+                self.logger.error("Failed to populate cve: %s products: %s", cve, products_copy)
 
-        for product in products:
-            status_id = product[2]
-            product_data = (product[0], product[1], product[3])
-            products_by_status[status_id].append(product_data)
-
-        return products_by_status
-
-    def _populate_cves(self, csaf_cves: CsafCves):
-        for cve in csaf_cves.keys():
-            cve_products = csaf_cves.to_tuples(cve, ("cpe", "package", "status_id", "module"))
-            products = self._map_products_by_status(cve_products)
-            to_insert, to_associate_ids = self._get_product_rows(products)
-            inserted_ids = self._insert_missing_products(to_insert)
-            all_ids = to_associate_ids+inserted_ids
-            self._remove_cves(cve, all_ids)
-            self._insert_cves(cve, all_ids)
-
-    def _delete_unreferenced_products(self):
+    def _delete_unreferenced_products(self) -> None:
+        cur = self.conn.cursor()
         try:
-            cur = self.conn.cursor()
-            cur.execute("""DELETE FROM csaf_products
+            cur.execute(
+                """DELETE FROM csaf_product
                             WHERE id IN (
                                 SELECT p.id
-                                FROM csaf_products AS p
-                                LEFT JOIN csaf_cves AS c ON c.csaf_product_id = p.id
-                                WHERE c.csaf_product_id IS NULL)""")
+                                FROM csaf_product AS p
+                                LEFT JOIN csaf_cve_product AS c ON c.csaf_product_id = p.id
+                                WHERE c.csaf_product_id IS NULL)"""
+            )
             self.conn.commit()
         except Exception as exc:
             CSAF_FAILED_DELETE.inc()
-            self.logger.exception("Failure while deleting data from csaf_products: %s", exc)
+            self.logger.exception("%s: ", FAILED_PRODUCT_DELETE)
             self.conn.rollback()
+            raise CsafStoreException(FAILED_PRODUCT_DELETE) from exc
         finally:
             cur.close()
 
-    def store(self, csaf_data: CsafData):
+    def store(self, csaf_data: model.CsafData) -> None:
         """Store collection of CSAF files into DB."""
         self._save_csaf_files(csaf_data.files)
```

Comment:
Not really related to the PR but could the `updated` timestamp be updated for each file separately and only once the CVE data are succefully synced (no error occurs)?

Reply:
sure why not